### ISSUE TITLE
Export Prometheus metrics

### ISSUE THREAD

**Issue by skitt** [submariner-io/submariner-operator#532]
On top of the existing Prometheus framework, export:

* number of configured gateways
* number of connected tunnels
* tx/rx bytes

**Comment by skitt** [comment]
Blocked by #533.

**Comment by mangelajo** [comment]
number of failing tunnels 
tx bytes by tunnel
rx bytes by tunnel
total tx bytes
total rx bytes
latency histogram

**Comment by skitt** [comment]
Version 0.5 now has:

* number of gateways
* number of reconciliations (useful for us, I think we’re doing too many)
* number of connections, with their status

**Comment by skitt** [comment]
Our telemetry experts recommend instrumenting as close to the event source as possible, so we should move the connection metrics to the gateway pods instead of sending them to the operator.

**Comment by mangelajo** [comment]
> Our telemetry experts recommend instrumenting as close to the event source as possible, so we should move the connection metrics to the gateway pods instead of sending them to the operator.

makes sense, may be some of the metrics can come from the operator (like detecting failing pods, etc... ) ...

**Comment by nyechiel** [comment]
@roytman FYI, these metrics are cable-driver specific. You may want to enhance the Wireguard implementation to support it

**Comment by nyechiel** [comment]
Closing as a duplicate of https://github.com/submariner-io/submariner-operator/issues/745
